### PR TITLE
Web Endpoint

### PR DESCRIPTION
**Pull request by RichardRohac**
**Added web endpoint API:**
- start by `export FLASK_APP=./WLC/web_endpoint.py; python3 -m flask run`

1. - there is a test webpage at `localhost:5000/api/upload_image` where you can try uploading code image; it returns JSON with format: [unfixed_code, fixed_code, result, error]
2. - also there is a test webpage at `localhost:5000/api/resubmit_code` where you can test resubmission; this one returns JSON [result, error]

- a.k.a. it accepts POST with some data, returns json

Also, refactored benchmark and code executor so there's no duplication between GUI, CLI and Web.

### CODE REVIEW

File: WLC/utils/gui.py [hunk 1]
```diff
@@ -23,28 +20,31 @@ def init_button(self):
...
-    def display_ocr(self, code):
+    def display_ocr(self, unfixed_code, fixed_code):
         text = tk.Text(self.master, height=14, width=70)
         text.pack(side=tk.LEFT, fill=tk.Y)
-        text.insert(tk.INSERT, "CODE READ BY OCR : \n")
-        text.insert(tk.END, code)
+        text.insert(tk.INSERT, "UNFIXED CODE READ BY OCR: \n")
+        text.insert(tk.END, unfixed_code)
+
+        text.insert(tk.INSERT, "\nCODE AFTER FIXING: \n")
```
Comment (SuNaden) [inline WLC/utils/gui.py]: Do we want to stick both into same text field?

**Review by SuNaden — CHANGES_REQUESTED** [review]
Just one thing, otherwise looks good to me :shipit:

File: WLC/utils/gui.py (hunk 1 above)
Comment (RichardRohac) [reply WLC/utils/gui.py]: Well that is up to you, if we plan user to edit the code and resubmit even in GUI then we need to split these (as user would be editing fixed code only)

Comment (RichardRohac) [reply WLC/utils/gui.py]: This is done now.

**Review by SuNaden — APPROVED** [review]

**Comment by SuNaden** [comment]
👍

**Pull request merged**